daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -798,6 +798,26 @@ class IdeLedgerClient(
         Left(fromScenarioError(err))
     }
 
+  override def trySubmitConcurrently(
+      actAs: OneAnd[Set, Ref.Party],
+      readAs: Set[Ref.Party],
+      commandss: List[List[command.ApiCommand]],
+      optLocation: Option[Location],
+  )(implicit
+      ec: ExecutionContext,
+      mat: Materializer,
+  ): Future[Seq[Either[SubmitError, Seq[ScriptLedgerClient.CommandResult]]]] =
+    // Since the IDE ledger doesn't have a sequencer, we simply lie about the
+    // concurrent part and sequence the commandss manually.
+    commandss.foldLeft(
+      Future.successful(Seq.empty[Either[SubmitError, Seq[ScriptLedgerClient.CommandResult]]])
```

Comment:
Avoid to use Seq.empty (and Seq(...)) when you need to instantiate a Seq. Pick a concrete sub-type, e.g. (List, Vector, ...).  This will make cleared what is the complexity of your code.
Here you will get actually a  `List`, and end up with a quadratic complexity as you append a element a the end of the list at each iteration. 

More generally, I will arg we problably should replace most of the `Seq` in the LedgerClient by a concrete type, but this is outside the scope of your work.

Reply:
I didn't know this! Thanks Remy!

Reply:
thanks for spotting this. I've replaced the outer `Seq` in the return type of `trySubmitConcurrently` with `List` and for this implementation I changed the fold to use a `ListBuffer` as the accumulator